src/content/ja/2019/compression.md

```diff
@@ -0,0 +1,321 @@
+     <td><p style="text-align: right">0.000007%</p></td>
+     <td><p style="text-align: right">33</p></td>
+     <td><p style="text-align: right">32</p></td>
+    </tr>
+    <tr>
+     <td>x-compress</td>
+     <td><p style="text-align: right">0.000002%</p></td>
+     <td><p style="text-align: right">0.000006%</p></td>
+     <td><p style="text-align: right">8</p></td>
+     <td><p style="text-align: right">29</p></td>
+    </tr>
+  </table>
+  <figcaption>図1.圧縮アルゴリズムの採用。</figcaption>
+</figure>
+
+圧縮されて提供されるリソースの大半は、gzip（80％）またはbrotli（20％）のいずれかを使用しています。他の圧縮アルゴリズムはあまり使用されません。
+
+<figure>
+  <a href="/static/images/2019/15_Compression/fig2.png">
+    <img src="/static/images/2019/15_Compression/fig2.png" alt="図2.デスクトップページでの圧縮アルゴリズムの採用。" aria-labelledby="fig2-caption" aria-describedby="fig2-description" width="600" height="371" data-width="600" data-height="371" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vQNIyMEGYE_1W0OdFYLIKsxg6M3o_ZsTTuaX73Zzv6Alw4x4D6oH0jdg9BSgw-jy4E-MmX_Qaf-B98W/pubchart?oid=2052550005&amp;format=interactive">
+  </a>
+  <div id="fig2-description" class="visually-hidden">図1のデータテーブルの円グラフ。</div>
+  <figcaption id="fig2-caption">図2.デスクトップページでの圧縮アルゴリズムの採用。</figcaption>
+</figure>
+
+さらに「none」「UTF-8」「base64」「text」など、無効な`Content-Encoding`を返す67Kのリクエストがあります。これらのリソースは圧縮されていない状態で提供される可能性があります。
+
+HTTPアーカイブによって収集された診断から圧縮レベルを判断することはできませんが、コンテンツを圧縮するためのベストプラクティスは次のとおりです。
+
+*   少なくとも、テキストベースのアセットに対してgzip圧縮レベル6を有効にします。これは、計算コストと圧縮率の間の公平なトレードオフを提供し、[多くのWebサーバーのデフォルト](https://paulcalvano.com/index.php/2018/07/25/brotli-compression-how-much-will-it-reduce-your-content/)にもかかわらず、[Nginxは依然として低すぎることが多いレベル1のままです](http://nginx.org/en/docs/http/ngx_http_gzip_module.html#gzip_comp_level)。
+*   brotliおよびprecompressリソースをサポートできる場合は、brotliレベル11に圧縮します。これはgzipよりも計算コストが高くなります。したがって、遅延を避けるためには、事前圧縮が絶対に必要です。
+*   brotliをサポートでき、事前圧縮できない場合は、brotliレベル5に圧縮します。このレベルでは、gzipと比較してペイロードが小さくなり、同様の計算オーバーヘッドが発生します。
+
+
+## どの種類のコンテンツを圧縮していますか？
+
+ほとんどのテキストベースのリソース（HTML、CSS、JavaScriptなど）は、gzipまたはbrotli圧縮の恩恵を受けることができます。ただし、多くの場合、これらの圧縮技術をバイナリリソースで使用する必要はありません。画像、ビデオ、一部のWebフォントなどが既に圧縮されているため。
+
+次のグラフでは、上位25のコンテンツタイプが、リクエストの相対数を表すボックスサイズで表示されています。各ボックスの色は、これらのリソースのうちどれだけ圧縮されて提供されたかを表します。ほとんどのメディアコンテンツはオレンジ色で網掛けされていますが、これはgzipとbrotliにはほとんどまたはまったく利点がないためです。テキストコンテンツのほとんどは、それらが圧縮されていることを示すために青色で網掛けされています。ただし、一部のコンテンツタイプの水色の網掛けは、他のコンテンツタイプほど一貫して圧縮されていないことを示しています。
+
+<figure>
+  <a href="/static/images/2019/15_Compression/fig3.png">
+    <img src="/static/images/2019/15_Compression/fig3.png" alt="図3.上位25の圧縮コンテンツタイプ。" aria-labelledby="fig3-caption" aria-describedby="fig3-description" width="780" height="482" data-width="780" data-height="482" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vQNIyMEGYE_1W0OdFYLIKsxg6M3o_ZsTTuaX73Zzv6Alw4x4D6oH0jdg9BSgw-jy4E-MmX_Qaf-B98W/pubchart?oid=1790056981&amp;format=interactive">
```

Comment:
In #683 we changed all the image directory folders to can you change all instances of `15_Compression` to `compression`?

Example changes needed below - can you do the rest as will be easier to do in one go than with loads of GitHub suggestions?). Note you have to change in two places:

```suggestion
  <a href="/static/images/2019/compression/fig3.png">
    <img src="/static/images/2019/compression/fig3.png" alt="図3.上位25の圧縮コンテンツタイプ。" aria-labelledby="fig3-caption" aria-describedby="fig3-description" width="780" height="482" data-width="780" data-height="482" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vQNIyMEGYE_1W0OdFYLIKsxg6M3o_ZsTTuaX73Zzv6Alw4x4D6oH0jdg9BSgw-jy4E-MmX_Qaf-B98W/pubchart?oid=1790056981&amp;format=interactive">
```

Reply:
Thanks, I fix.